src/World.h

```diff
@@ -279,8 +279,8 @@ class cWorld :
 	/** Finds a player from a partial or complete player name and calls the callback - case-insensitive */
 	bool FindAndDoWithPlayer(const AString & a_PlayerNameHint, cPlayerListCallback a_Callback);  // >> EXPORTED IN MANUALBINDINGS <<
 
-	// TODO: This interface is dangerous - rewrite to DoWithClosestPlayer(pos, sight, action)
-	cPlayer * FindClosestPlayer(Vector3d a_Pos, float a_SightLimit, bool a_CheckLineOfSight = true);
+	/** Calls the callback for closest player for given position, return true if player was found and callback called, false if player not found. */
+	bool DoWithClosestPlayer(Vector3d a_Pos, float a_SightLimit, cPlayerListCallback a_Callback, bool a_CheckLineOfSight = true, bool a_IgnoreSpectator = true);
```

Comment:
I think `DoWithNearestPlayer` is a better name.

Reply:
Also, if the player is found, this function should return whatever value the callback returned (all our callback-based `DoWith` functions work this way)

Reply:
`a_SightLimit` should probably be renamed to `a_RangeLimit`, because it wouldn't make sense if `a_CheckLineOfSight` was `false`.